qiskit/circuit/quantumcircuit.py

```diff
@@ -285,6 +286,45 @@ def __init__(
             raise TypeError("Only a dictionary or None is accepted for circuit metadata")
         self._metadata = metadata
 
+    @staticmethod
+    def from_instructions(
```

Comment:
A minor query: is there a chance that a subclass of `QuantumCircuit` might want to use this method as its own alternate constructor?  If so, perhaps this could be a `classmethod` that does the construction by `cls(name=..., ...)` instead of hard-coding `QuantumCircuit`?  I'm not 100% sure that there are any sensible subclasses of `QuantumCircuit` (the class certainly isn't constructed with co-operative subclassing in mind), so feel free to dismiss this.

Reply:
Wouldn't that raise an error if the subclass has a different signature for its `__init__`?

Reply:
It would, but that's the responsibility of a subclass to fix if it's modifying the signature of certain methods.  I don't have a strong opinion here - it's easy for us to change to the new form in the future, if it ever becomes useful to have it as a classmethod rather than a staticmethod.

In more sane OO languages it's not so much of an issue, because you're usually more forced to make decisions about which methods are virtual, so designing classes for co-operative subclassing is more at the fore-front.  Python's just a bit special in that regard.

Reply:
Okay, let's just keep it as is for now, and change it in the future if the need arises.